KK-Repos/Instill_Project
# Added automation test cases for get_word_meaning endpoint

# 🚀 Instill QA Test Suite 🚀
This repository contains an automation test suite for the Instill Project, implemented using the *PyTest library.*

Test Suite Execution Guide
1. Install Dependencies
	Ensure that you have the required dependencies by executing the following command in your terminal:
	 `pip install -r requirements.txt`
2. Start the Backend Server
Launch the backend server using the command:
       `python3 start_server.py`

Wait for the server to start successfully before proceeding.

3. Run PyTest Test Suite
Execute the following command to run the PyTest test suite:
	`pytest test/test_instill_qa.py`

## ⚡️ Additional Information
**Bug Fix: Duplicate Words in Payload**

A bug related to duplicate words in the payload has been fixed. Refer to the GitHub issue for details. - click [here](https://github.com/KK-Repos/Instill_Project/issues/2)

After - Fixed 

![image](https://github.com/KK-Repos/Instill_Project/assets/55937175/149ad1dd-b098-4c59-bcc5-9a8a4c9c73f4)


New Test Cases Added with combination positive and negative

Test cases have been added for the 'get-word-meaning' endpoint in the post method.

(Attach the screenshot that showcases the successful execution of the test suite for  reference.)

![image](https://github.com/KK-Repos/Instill_Project/assets/55937175/d8c3e251-8db0-4fc3-a2ff-bd797198130a)
